numba/np/arrayobj.py

```diff
@@ -1423,24 +1423,32 @@ def ol_numpy_broadcast_shapes(*args):
                    f'Got {arg}')
             raise errors.TypingError(msg)
 
-    def impl(*args):
-        # discover the number of dimensions
-        m = 0
-        for val in literal_unroll(args):
-            if isinstance(val, int):
-                m = max(m, 1)
-            else:
-                m = max(m, len(val))
-
-        # propagate args
-        r = [1] * m
-        for arg in literal_unroll(args):
-            if isinstance(arg, tuple) and len(arg) > 0:
-                numpy_broadcast_shapes_list(r, m, arg)
-            elif isinstance(arg, int):
-                numpy_broadcast_shapes_list(r, m, (arg,))
-        return r
-    return impl
+    # discover the number of dimensions
+    m = 0
+    for arg in literal_unroll(args):
```

Comment:
```suggestion
    for arg in args:
```
This is pure python code?

Reply:
Done!